packages/azure-kusto-data/source/tokenProvider.ts

```diff
@@ -97,6 +97,16 @@ abstract class MsalTokenProvider extends TokenProviderBase {
         this.authorityId = authorityId;
     }
 
+    commonOptions(): Configuration {
+        return {
+            auth: {
+                clientId: this.cloudInfo.KustoClientAppId,
```

Comment:
Please only set this in user based auth methods

Reply:
clientId is a mandatory parameter, so the fix is a bit akward but done.